src/gui/GuiEvents_TEST.cc

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include "ignition/gazebo/test_config.hh"
+#include "ignition/gazebo/gui/GuiEvents.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace gui;
+
+/////////////////////////////////////////////////
+TEST(GuiEventsTest, GuiNewRemovedEntities)
+{
+  events::GuiNewRemovedEntities event({1, 2, 3}, {4, 5, 6});
+
+  EXPECT_LT(QEvent::User, event.type());
+
+  auto addedEntities = event.NewEntities();
```

Comment:
```suggestion
  auto addedEntities = event.NewEntities();
  EXPECT_EQ(3u, addedEntities.size());
```
nit: we can add a check to verify that there are exactly 3 added entities

Reply:
Good idea, see 43a014f765088c34157d9c4caecff6cb77fe4242